### PR TITLE
SO-5868: Retrieve validation results via REST

### PR DESCRIPTION
**Pull request by adamfilep**

### CODE REVIEW

**Review by apeteri — APPROVED** [review]
🥇

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java [hunk 1]
```diff
@@ -181,6 +181,16 @@ public SnowOwlOpenApiWebMvcResource openApiWebMvcResource(
 	@Bean
 	@Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
 	public ObjectMapper objectMapper(@Autowired HttpServletRequest request) {
+		return objectMapperFromRequest(request);
```
Comment (cmark) [inline core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:185]: I think for consistency, this should be a provider as well.

**Review by cmark — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java (hunk 1 above)
Comment (adamfilep) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:185]: As I see it, we are going to need the two different methods:
This method with ObjectMapper return type is necessary for mappingJackson2HttpMessageConverter()
And the other one, with Provider<ObjectMapper> is necessary for the RepositoryValidationRestService class

Comment (apeteri) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:185]: IIRC we tried three approaches:

* Keep the initialization code in this method and refer to it in the provider lambda
    * &rarr; the `Provider` produced `ObjectMapper` delegates (proxies) that expected a request scope to be present, triggering the exception in the validation REST controller (where it was originally reported)
* Move the initialization code to the lambda and create an instance here using the `Provider`
    * &rarr; the returned instance is an actual `ObjectMapper` but creating it requires a request scope which is not available at startup, triggering the exception right away
* Move the initialization code to a non-&commat;`Bean`-annotated method and refer to it from both annotated methods
    * &rarr; the current state of the PR that worked without issues

**Review by cmark — APPROVED** [review]
🥓

**Pull request merged**